sim/druid/insect_swarm.go

```diff
@@ -0,0 +1,70 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (druid *Druid) registerInsectSwarmSpell() {
+	hasGlyph := druid.HasPrimeGlyph(proto.DruidPrimeGlyph_GlyphOfInsectSwarm)
+
+	druid.InsectSwarm = druid.RegisterSpell(Humanoid|Moonkin, core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 5570},
+		SpellSchool:    core.SpellSchoolNature,
+		ProcMask:       core.ProcMaskSpellDamage,
+		ClassSpellMask: DruidSpellInsectSwarm | DruidNatureSpells | DruidSpellDoT,
```

Comment:
The ClassSpellMask should be only the spell mask. The group masks already contain your spell mask inside them (along with a bunch of other spells masks) so you are basically assigning all those masks to this spell instead of just its own.